src/app/public/modules/i18n/currency/currency-format.service.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  SkyCurrencyFormat, SkyCurrencySymbolLocation
+} from './currency-format';
+
+const DEFAULT_LOCALE = 'en-US';
+const DEFAULT_CURRENCY_CODE = 'USD';
+const DEFAULT_GROUP_CHARACTER = ',';
+const DEFAULT_DECIMAL_CHARACTER = '.';
+
+/**
+ * Service to get a currency's format given an iso code.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class SkyCurrencyFormatService {
+  /**
+   * Gets a currency's format.
+   * @param params optional params object
+   * @param params.isoCurrencyCode the ISO 4217 Currency Code. Defaults to `USD`.
+   * @param params.locale the locale. Defaults to `en-US`. Examples: `en-US`, `en-GB`, `fr-FR`.
+   */
+  public getCurrencyFormat(params?: Partial<IsoCurrencyCodeAndLocale>): SkyCurrencyFormat {
+    const isoCurrencyCode = params?.isoCurrencyCode ?? DEFAULT_CURRENCY_CODE;
+    const locale = params?.locale ?? DEFAULT_LOCALE;
+    const formatter = new Intl.NumberFormat(locale, {
+      style: 'currency',
+      currency: isoCurrencyCode
+    });
+
+    const resolvedOptions: Intl.ResolvedNumberFormatOptions = formatter.resolvedOptions();
+    const currencyCode = resolvedOptions.currency;
+    const parts = this.formatToParts(formatter);
+
+    return {
+      locale: locale,
+      isoCurrencyCode: currencyCode,
+      symbol: parts.symbol,
+      symbolLocation: parts.symbolLocation,
+      decimalCharacter: parts.decimalCharacter,
+      groupCharacter: parts.groupCharacter,
+      precision: resolvedOptions.maximumFractionDigits
+    };
+  }
+
+  /**
+   * Given an ISO 4217 Currency Code return the currency's precision.
+   * @param isoCurrencyCode the ISO 4217. Defaults to `USD`.
+   */
+  public getCurrencyPrecision(isoCurrencyCode: string): number {
+    return this.getCurrencyFormat({ isoCurrencyCode: isoCurrencyCode }).precision;
+  }
+
+  private formatToParts(formatter: Intl.NumberFormat): CurrencyFormatParts {
+    const BIG_VALUE_TO_GET_PART_INFO: number = 100_000_000;
+
+    if (formatter.formatToParts) {
+      const parts = formatter.formatToParts(BIG_VALUE_TO_GET_PART_INFO);
+
+      type IntlFindFn = (intlType: Intl.NumberFormatPartTypes, defaultValue: string) => string;
+      const findOrDefault: IntlFindFn = (intlType, defaultValue) =>
+        parts.find(p => p.type === intlType)?.value ?? defaultValue;
+
+      return {
+        symbol: findOrDefault('currency', ''),
+        symbolLocation: parts.findIndex(p => p.type === 'currency') === 0 ? 'prefix' : 'suffix',
+        decimalCharacter: findOrDefault('decimal', DEFAULT_DECIMAL_CHARACTER),
+        groupCharacter: findOrDefault('group', DEFAULT_GROUP_CHARACTER)
+      };
+    }
+
+    return this.shimFormatToParts(formatter);
+  }
+
+  /**
+   * Shims INTL.NumberFormatter.formatToParts since it does not exist in IE.
+   * @param isoCurrencyCode
+   * @param locale
+   */
+  private shimFormatToParts(formatter: Intl.NumberFormat): CurrencyFormatParts {
```

Comment:
So I found out that Angular's `GetCurrencySymbol` does require Angular locales 👎 

I spent a bit thinking on how to handle this shim without importing a library or defaulting the symbol to something non-useful and landed on this.

**General idea:** _(see table below for step-by-step example)_
1. Use `NumberFormatter.format` to format to a known currency value of 0. 
2. Remove the numeric section from the value and then trim the whitespace.
3. Result: You now have your currency symbol.

Pros:
- We return the exact symbol that the INTL api supplies. 
- Since we have the exact currency symbol we can now use it to find whether its a prefix or suffix whereas we defaults to prefix before.
- We could _probably_ grab the exact decimal/group character. I wanted to vet this first before going further.

Cons:
- It is string manipulation 

**Examples of INTL.NumberFormatter.format(0)**
| Locale 	| Currency 	| format(0) 	| Numeric section 	| Result of removing numeric section (Currency Symbol) 	|
|--------	|----------	|-----------	|-----------------	|------------------------------------------------------	|
| en-US  	| USD      	| $0.00     	| 0.00            	| $                                                    	|
| en-CA  	| USD      	| US$0.00   	| 0.00            	| US$                                                  	|
| fr-CA  	| USD      	| 0,00 $ US 	| 0,00            	| $ US                                                 	|
| en-US  	| JPY      	| ¥0        	| 0               	| ¥                                                    	|
| en-CA  	| JPY      	| JP¥0      	| 0               	| JP¥                                                  	|
| fr-CA  	| JPY      	| 0 ¥       	| 0               	| ¥                                                    	|

Reply:
Oh nice. That stackoverflow is _pretty much_ doing the same thing as me but I like it better haha.

> It should be noted this is a somewhat naive approach, since it only works for number characters 0-9. You'd have to add any other number system you need to support in similar fashion.

- I was able to combine the post's answer and another thought I had to make it number system agnostic by using `(0).toLocaleString(locale);` to get the formatted zero (`0` for 0-9 systems or Arabic's `٠`). 
- Since we're formatting a currency string with only zeros for numbers I'm able to remove all formatted zeros to get the currency, decimal, and group character.